source/agora/consensus/EnrollmentManager.d

```diff
@@ -717,6 +717,58 @@ public class EnrollmentManager
         return result;
     }
 
+    /***************************************************************************
+
+        Get validator's pre-image at a certain height from the validator set
+
+        Params:
+            enroll_key = The key for the enrollment in which the pre-image is
+                contained.
+            height = The block height of the preimage to get
+            result_image = will contain the PreimageInfo if exists
+
+        Returns:
+            true if getting pre-image is successfully processed
+
+    ***************************************************************************/
+
+    public bool getValidatorPreimage (const ref Hash enroll_key, ulong height,
+        out PreimageInfo result_image) @safe nothrow
+    {
+        try
+        {
+            () @trusted {
```

Comment:
If nearly the entire body of the function has to be trusted, then you can mark `getValidatorPreimage` as @trusted, and therefore remove this inline delegate.